backend/controllers/playlist.controller.js

```diff
@@ -0,0 +1,130 @@
+import Playlist from '../models/playlists.model.js'
+import _ from 'lodash'
+
+//@route   GET api/playlist
+//@desc    Get All Playlists
+//@access  Private
+const getAllPlaylists = async (req, res) => {
+    try {
+        const id = req.user
+        const playlists = await Playlist.findOne({ user: id })
+        res.status(200).send(playlists)
+    } catch (err) {
+        res.status(500).json({ success: false, message: 'Server Error' })
+    }
+}
+
+
+//@route   PUT api/playlist
+//@desc    Toggle Playlists
+//@access  Private
+const togglePlaylist = async (req, res) => {
+    try {
+        const id = req.user
+        const { videoID, playlistID } = req.body
+        let lists = await Playlist.findOne({ user: id })
+        const updateItems = lists.playlists.map((p) => {
+            if (p._id == playlistID) {
+                const videoPresent = p.videos.some((v) => v == videoID)
+                if (videoPresent) {
+                    const filteredVideo = p.videos.filter((v) => v != videoID)
+                    return { "_id": p._id, "name": p.name, "videos": filteredVideo }
+                }
+                else {
+                    return { "_id": p._id, "name": p.name, "videos": [...p.videos, videoID] }
+                }
+            }
+            return p
+        })
+
+        const updateDetails = {
+            user: id,
+            playlists: updateItems
+        }
+        const final = await Playlist.findOneAndUpdate({ _id: lists._id }, { $set: updateDetails }, { new: true })
+        res.status(200).send(final)
+    } catch (err) {
+        res.status(500).json({ success: false, message: 'Server Error' })
```

Comment:
Har jagah repeat ho raha hai ye.. Since express does not support try catch by default, there is a way to put this in the errorHandler. Mere code mai hai..

https://github.com/gauravggg21/api-matcha-dev/blob/development/src/utils/asyncHandler.util.js

Reply:
Ohh, I really liked your approach of handling try catch. But if I change it right now in my code it can create a big mess for me
So, I will keep your approach in mind and will apply in my quiz app :)
Hope that's okay

Reply:
Yes yes, I was just putting it out there. :D